File: 06_retirement_calculator/python/retirement_calculator.py
import datetime


def get_current_age():
    return int(input("What is your current age? "))


def get_age_to_retire():
    return int(input("At what age would you like to retire? "))


def calculate_years_to_retire(age_to_retire, current_age):
    return age_to_retire - current_age


def get_current_year():
    now = datetime.datetime.now()
    return now.year


def print_years_to_retire(years_to_retire):
    print(f"You have {years_to_retire} years to retire.")


def print_year_of_retirement(current_year, years_to_retire):
    print(f"It's {current_year}, so you can retire in {current_year + years_to_retire}.")


def main():
    age_now = get_current_age()
    age_when = get_age_to_retire()
    year_now = get_current_year()
    years_mathed = calculate_years_to_retire(age_when, age_now)
    print_years_to_retire(years_mathed)
    print_year_of_retirement(year_now, years_mathed)


if __name__ == "__main__":
    main()

File: 06_retirement_calculator/javascript/retirement_calculator.js
var current_age_input = window.prompt("What is your current age?");
var age_to_retire_input = window.prompt("What age would you like to retire?");

var current_age = parseInt(current_age_input);
var age_to_retire = parseInt(age_to_retire_input);

var years_to_retire = (age_to_retire - current_age);

var today = new Date();
var year = today.getFullYear();

console.log(`You have about ${years_to_retire} years to retire.`);
console.log(`It's ${year}, so you can retire in ${year + years_to_retire}.`);
